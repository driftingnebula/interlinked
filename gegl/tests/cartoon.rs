use {
  gegl::{Cartoon, GeglOperation},
  insta::assert_debug_snapshot,
};

#[test]
fn test_cartoon_graph() {
  let mut graphs = vec![];

  let op = Cartoon {
    mask_radius: 0.1,
    ..Default::default()
  };

  let default = Cartoon::default();

  graphs.push(("non-default: exclude defaults", op.graph(false)));
  graphs.push(("non-default: include defaults", op.graph(true)));
  graphs.push(("default: exclude defaults", default.graph(false)));
  graphs.push(("default: include defaults", default.graph(true)));

  assert_debug_snapshot!("graphs", graphs);
}
