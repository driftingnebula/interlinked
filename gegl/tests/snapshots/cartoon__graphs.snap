---
source: gegl/tests/cartoon.rs
expression: graphs
---
[
    (
        "non-default: exclude defaults",
        [
            "gegl:cartoon",
            "mask_radius=0.1",
        ],
    ),
    (
        "non-default: include defaults",
        [
            "gegl:cartoon",
            "mask_radius=0.1",
            "pct_black=0.2",
        ],
    ),
    (
        "default: exclude defaults",
        [
            "gegl:cartoon",
        ],
    ),
    (
        "default: include defaults",
        [
            "gegl:cartoon",
            "mask_radius=7",
            "pct_black=0.2",
        ],
    ),
]
